Stop testing EOL Python 2.6 and 3.3

jdufresne opened the pull request:
Python 2.6 and 3.3 are EOL. They are no longer receiving bug fixes,
including for security issues. Python 2.6 went EOL on 2013-10-29 and
Python 3.3 went EOL on 2017-09-29. For details, see:

https://devguide.python.org/#status-of-python-branches

For additional discussion on why project maintainers may wish to drop
Python 2.6 support, please see:

https://snarky.ca/stop-using-python-2-6/
http://www.curiousefficiency.org/posts/2015/04/stop-supporting-python26.html

Reduces testing resources.

kjd commented:
My concern with a patch like this is that there are downstream packages like `requests` that depend on it and promise Python 2.6 support. I'd like to only deprecate the older version when there is no significant project relying on it in this fashion.

jdufresne commented:
I understand the concern. It is a fair one.

Requests also recently dropped 2.6 and 3.3 support. These changes will be released in requests 3.0. For the related PRs, see:

Dropped 2.6: https://github.com/requests/requests/pull/4118
Dropped 3.3: https://github.com/requests/requests/pull/4231
Release 3.0: https://github.com/requests/requests/pull/4546

Many other projects are dropping support for EOL Python versions. In my experience, there has been less and less Python 2.6 support. Projects wishing to stay on an older version of Python can continue to use idna by pinning the version in `requirements.txt` or the `install_requires` of `setup.py`. So the library will not be unusable to those projects.

Using [pypinfo](https://github.com/ofek/pypinfo), here are the download statistics for idna for the last 30 days by Python versions, showing 2.6 & 3.3 is a small fraction of the total:


| python_version | percent | download_count |
| -------------- | ------: | -------------: |
| 2.7            |  66.92% |      4,006,246 |
| 3.6            |  17.34% |      1,038,007 |
| 3.5            |   9.72% |        581,902 |
| 3.4            |   5.25% |        314,165 |
| 2.6            |   0.44% |         26,354 |
| 3.7            |   0.18% |         10,912 |
| 3.3            |   0.14% |          8,390 |
| 3.2            |   0.01% |            641 |
| None           |   0.00% |             89 |
| 3.8            |   0.00% |             41 |


This PR also adds `python_requires` to `setup.py`. For version of pip that support it, this will prevent pip from installing the project on unsupported Python versions. pip will instead choose a compatible version. pip has supported this since 9.0.

For more information, see:

https://packaging.python.org/tutorials/distributing-packages/#python-requires

Thanks for considering.

hugovk commented:
Reviewing the first 10 dependencies of IDNA from https://libraries.io/pypi/idna/usage:

| package | GH stars | 2.6 | 3.3
| -------------- | ------: | ------------- | ------------- |
| requests/requests     | 32201 | supported, next will drop | unsupported |
| tornadoweb/tornado    | 15697 | unsupported | unsupported |
| pyca/cryptography     | 2101  | unsupported | unsupported |
| certbot/certbot       | 22079 | unsupported | unsupported |
| docker/docker-py      | 2932  | unsupported | supported |
| msiemens/tinydb       | 2198  | unsupported | supported |
| lingui/js-lingui      | 1013  | unsupported | unsupported |
| buildbot/buildbot     | 3403  | unsupported | unsupported |
| HypothesisWorks/hypothesis | 2650  | unsupported | unsupported |
| blaze/odo             | 769 | unsupported | unsupported |
